### PR TITLE
fix: nonce zero

### PR DESCRIPTION
**Pull request by ashitakah**
# 🤖 Linear

Closes GRT-19

### CODE REVIEW

File: solidity/test/unit/Oracle.t.sol [hunk 1]
```diff
@@ -280,6 +280,41 @@ contract Oracle_Unit_CreateRequests is BaseTest {
...
+      mockRequest.requestModuleData = _requestData;
+      mockRequest.responseModuleData = _responseData;
+      mockRequest.disputeModuleData = _disputeData;
+      mockRequest.requester = requester;
+      mockRequest.nonce = uint96(0);
+
+      bytes32 _theoreticalRequestId = _getId(mockRequest);
+      _requests[_i] = mockRequest;
+      _precalculatedIds[_i] = _theoreticalRequestId;
+      _ipfsHashes[_i] = keccak256(abi.encode(_theoreticalRequestId, mockRequest.nonce));
```
Comment (0xJabberwock) [inline solidity/test/unit/Oracle.t.sol:309]: Given that all the requests in the batch are equal, the population of `mockRequest`, the calculation of `_theoreticalRequestId` and the `keccak256` encoding can be performed outside the `for` loop.

File: solidity/contracts/Oracle.sol [hunk 2]
```diff
@@ -378,10 +378,14 @@ contract Oracle is IOracle {
...
-  function _createRequest(Request calldata _request, bytes32 _ipfsHash) internal returns (bytes32 _requestId) {
+  function _createRequest(Request memory _request, bytes32 _ipfsHash) internal returns (bytes32 _requestId) {
     uint256 _requestNonce = totalRequestCount++;
 
-    if (_requestNonce != _request.nonce || msg.sender != _request.requester) revert Oracle_InvalidRequestBody();
+    if (msg.sender != _request.requester || (_requestNonce != _request.nonce && _request.nonce != 0)) {
+      revert Oracle_InvalidRequestBody();
+    }
+
+    if (_request.nonce == 0) _request.nonce = uint96(_requestNonce);
```
Comment (0xJabberwock) [inline solidity/contracts/Oracle.sol:388]: If you put the last `if` above then you wouldn't need to modify the reversion line.

Comment (ashitakah) [reply solidity/contracts/Oracle.sol]: 👍

File: solidity/test/unit/Oracle.t.sol (hunk 1 above)
Comment (ashitakah) [reply solidity/test/unit/Oracle.t.sol]: 👍

**Review by 0xJabberwock — APPROVED** [review]

**Review by 0xJabberwock — APPROVED** [review]
Approval granted since tests pass locally. 👍

**Review by 0xShaito — APPROVED** [review]

**Pull request merged**